### DISCUSSION TITLE
All routers in a network

### DISCUSSION THREAD

**Discussion by rprj** [openthread/openthread#8944] [Q&A] [answered]
If I don't want any _child_ in my network, is it possible to make all the full-thread devices in a network to have _routers_ as their node states? How do I go about doing that? Thank you!

**Comment by abtink** [comment]
Every  node that attaches to Thread mesh (except the very first node), attaches as a `child` first but it can later promote to `router` (if it is FTD).

The decision on when and which nodes promote to `router` is controlled by the Thread mesh network itself.  Thread spec defines  the rules related to this (when/how to upgrade or downgrade). For example, check Section 5.9.9 titled "Router ID Management" in Thread 1.3.0 spec.
- In short, Thread mesh tries to keep the number routers in range from 16 (`ROUTER_UPGRADE_THRESHOLD`) up to ~23.
- So if you have less than 16 FTD nodes  in your network, they should all eventually promote to router role.

**Reply by rprj** [reply]
If I have ~100 FTD nodes in my network and I want them all to be routers, are there any parameters in the code I can change to do that? Even if they create multiple sub-networks of different partitions as long as every single node is a router.

**Reply by abtink** [reply]
Thread spec limits the number of active router in a Thread mesh to max of `32`. Other FTD device can take a `REED` role (router-eligible end-device). This is to ensure "routing info" tracked by each router is kept small and manageable (this is used for routing traffic within mesh).

Any reason why you want all devices to be `router`? I don't see any benefit behind this.

**Reply by rprj** [reply]
I just want each unit to be able to forward packets in multiple directions and not be limited by one path if their node state is child.

Is making all devices strictly router only possible?

**Reply by jwhui** [reply]
Thread Specification states a hard limit of 32 active routers in a given Thread Partition. You could try increasing that number beyond 32, but you would no longer be conformant to the Thread Specification.

As previously stated, the reason to limit the number of routers is to bound the state and communication complexity related to maintaining routing tables. Thread also limits the number of routers to improve multicast performance (since only routers will forward multicast packets).

**Reply by rprj** [reply]
Wouldn't different Thread partitions be able to communicate with each other? Or would that cause issues?

**Reply by jwhui** [reply]
Thread Partitions form because they do not have direct radio connectivity. If two partitions do suddenly find themselves within radio connectivity, they will merge into a single partition.

**Reply by rprj** [reply]
Ah, I see.

What about if I wanted to make some devices strictly router? Would doing 'state router' suffice?

**Reply by jwhui** [reply, marked as answer]
In short, no. Thread devices all attach initially as an End Device. The only exception is when becoming a Leader to start a new partition. Thread also specifies clear rules on when an End Device may upgrade to the Route role - see the [Thread Specification](https://www.threadgroup.org/support#specifications) for full details.

**Reply by rprj** [reply]
Thanks
